contracts/voting-verifier/src/execute.rs

```diff
@@ -171,21 +230,40 @@ pub fn vote(
 ) -> Result<Response, ContractError> {
     let poll = POLLS
         .may_load(deps.storage, poll_id)?
-        .ok_or(ContractError::PollNotFound)?
-        .try_map(|poll| {
-            poll.cast_vote(env.block.height, &info.sender, votes)
-                .map_err(ContractError::from)
-        })?;
+        .ok_or(ContractError::PollNotFound)?;
 
+    let initial_results = get_indices_with_quorum(&poll);
+
+    let poll = poll.try_map(|poll| {
+        poll.cast_vote(env.block.height, &info.sender, votes)
+            .map_err(ContractError::from)
+    })?;
     POLLS.save(deps.storage, poll_id, &poll)?;
 
-    Ok(Response::new().add_event(
-        Voted {
-            poll_id,
-            voter: info.sender,
-        }
-        .into(),
-    ))
+    let final_results = get_indices_with_quorum(&poll);
+
+    // we only want to emit a quorum reached event when quorum is first reached, and not subsequently
+    let reached_quorum: Vec<&(u32, Vote)> = final_results
```

Comment:
I am so confused looking at these variable names. `initial_results` and `final_results` coming from a function called `get_indices_with_quorum`, and then `reached_quorum` being a `Vec` with its name sounds liked a boolean.

Reply:
yeah I really struggled with finding good names for these variables. Do you have any suggestions?

Reply:
I need to understand what's going on here before I am able to suggest anything.